app/controllers/katello/api/v2/api_controller.rb

```diff
@@ -86,7 +86,7 @@ def item_search(item_class, param_hash, options)
     end
 
     def scoped_search(query, default_sort_by, default_sort_order, options = {})
-      resource = options.fetch(:resource_class, resource_class)
+      resource = options.fetch(:resource_class) || resource_class
```

Comment:
If `resource_class` is not provided, this will error out before it gets to the `||` I believe.